app/commands.py

```diff
@@ -664,17 +658,25 @@ async def _map(ctx: Context) -> str | None:
                 )
             ]
 
+            # update all maps in the set
+            for bmap in bmap.set.maps:
+                await maps_repo.update(bmap.id, status=new_status, frozen=True)
+
+            # make sure cache and db are synced about the newest change
             for bmap in app.state.cache.beatmapset[bmap.set_id].maps:
                 bmap.status = new_status
+                bmap.frozen = True
 
         else:
             # update only map
             await maps_repo.update(bmap.id, status=new_status, frozen=True)
 
             map_ids = [bmap.id]
 
+            # make sure cache and db are synced about the newest change
             if bmap.md5 in app.state.cache.beatmap:
                 app.state.cache.beatmap[bmap.md5].status = new_status
+                app.state.cache.beatmap[bmap.md5].frozen = True
```

Comment:
nit: can both if/else branches be adjusted so `map_ids` is fetched last?

e.g.
```py
update_sql()
update_cache()
map_ids = ...
```

Reply:
it's kinda a side effect of the real behaviour of the branches so it's secondary